projects/novo-elements/src/elements/data-table/data-table.component.ts

```diff
@@ -459,14 +458,22 @@ export class NovoDataTable<T> implements AfterContentInit, OnDestroy {
 
   @HostBinding('class.empty')
   get empty() {
-    return this.dataSource && this.dataSource.totallyEmpty;
+    if (this.useOverrideTotal) {
+      return this.overrideTotal === 0;
+    } else {
+      return this.dataSource && this.dataSource.totallyEmpty;
+    }
```

Comment:
Just for the sake of brevity and readability, you could do this:

```
return this.useOverrideTotal ? this.overrideTotal === 0 : this.dataSource?.totallyEmpty;
```

Reply:
Unit Tests?

Reply:
updated with simplified logic, and added unit tests